### PR TITLE
refactor: Remove core config from dashboard tenants list response

### PR DESCRIPTION
**Pull request by KShivendu**
## Summary of change

Remove core config from dashboard tenants list response

## Related issues

-   https://github.com/supertokens/supertokens-node/pull/635/
## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py
-   [ ] Make sure that `syncio` / `asyncio` functions are consistent.

### CODE REVIEW

File: supertokens_python/recipe/multitenancy/interfaces.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def __init__(
         emailpassword: EmailPasswordConfig,
         passwordless: PasswordlessConfig,
         third_party: ThirdPartyConfig,
-        core_config: Dict[str, Any],
+        core_config: Optional[Dict[str, Any]],
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/multitenancy/interfaces.py:104]: this should not be changed optional... you can make a different type for this in the dashboard recipe which will NOT have this core_config prop and use that in the dashboard recipe. But do not change the mulittenancy recipe at all.

File: supertokens_python/recipe/multitenancy/interfaces.py [hunk 2]
```diff
@@ -126,13 +126,16 @@ def __init__(
...
+        res = {
             "tenantId": self.tenant_id,
             "emailpassword": self.emailpassword.to_json(),
             "passwordless": self.passwordless.to_json(),
             "thirdParty": self.third_party.to_json(),
-            "coreConfig": self.core_config,
         }
+        if self.core_config is not None:
+            res["coreConfig"] = self.core_config
+
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/multitenancy/interfaces.py:137]: undo this change

File: supertokens_python/recipe/multitenancy/recipe_implementation.py [hunk 3]
```diff
@@ -193,7 +193,7 @@ async def list_all_tenants(
                 config.emailpassword,
                 config.passwordless,
                 config.third_party,
-                config.core_config,
+                config.core_config or {},
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/multitenancy/recipe_implementation.py:196]: undo this change

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/recipe/multitenancy/interfaces.py (hunk 2 above)
Comment (KShivendu) [reply supertokens_python/recipe/multitenancy/interfaces.py]: Done

File: supertokens_python/recipe/multitenancy/interfaces.py (hunk 1 above)
Comment (KShivendu) [reply supertokens_python/recipe/multitenancy/interfaces.py]: DOne

File: supertokens_python/recipe/multitenancy/recipe_implementation.py (hunk 3 above)
Comment (KShivendu) [reply supertokens_python/recipe/multitenancy/recipe_implementation.py]: Done
